src/dealer_router.rs

```diff
@@ -74,8 +75,7 @@ impl SocketBackend for RouterSocketBackend {
 
 pub struct RouterSocket {
     backend: Arc<RouterSocketBackend>,
-    _accept_close_handle: Option<oneshot::Sender<bool>>,
-    binds: Vec<Endpoint>,
+    binds: HashMap<Endpoint, oneshot::Sender<bool>>,
 }
 
 impl Drop for RouterSocket {
```

Comment:
Do we need to signal the channel on shutdown? If not, could you add a oneline comment explaining why not? Also as a probably unrelated question, does backend.shutdown() do anything that simply dropping the backend doesnt? Is there a reason you would want to shut the backend down without dropping it?

Reply:
> Also as a probably unrelated question, does backend.shutdown() do anything that simply dropping the backend doesnt? Is there a reason you would want to shut the backend down without dropping it?

Droping backend will not stop everything properly. Since backend is kept inside by an `Arc` other references may remain preventing it from closing. Main idea behind `shutdown` call is that it's built in such a way that it notifies other coroutines holding references to a backend that they should stop working.

> Do we need to signal the channel on shutdown? If not, could you add a oneline comment explaining why not?

Basically oneshot::Sender is built in such a way that it notifies a Receiver part when it's dropped. So there is no need to signal it explicitly.